Assets/Scripts/Puzzles/Color/Player.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DN.Puzzle.Color
+{
+	/// <summary>
+	/// This player class is unique to the Color puzzle, and follows
+	/// all the lines that are available in the current node if specified by
+	/// the queue.
+	/// </summary>
+	public class Player : MonoBehaviour
+	{
+		public enum State
+		{ 
+			Idle,
+			Navigating,
+			Moving,
+			Waiting,
+			Stuck,
+			Done,
+		}
```

Comment:
public enum  misschien in apart script dan kan iedereen die states gebruiken

Reply:
Iedereen kan de states al gebruiken (`Player.State`) De benaming van deze enum is te general om in een apart script te gooien.